src/items.rs

```diff
@@ -607,6 +607,112 @@ pub fn format_struct(context: &RewriteContext,
     }
 }
 
+pub fn format_trait(context: &RewriteContext, item: &ast::Item, offset: Indent) -> Option<String> {
+    if let ast::ItemKind::Trait(unsafety, ref generics, ref type_param_bounds, ref trait_items) =
+           item.node {
+        let mut result = String::new();
+        let header = format!("{}{}trait {}",
+                             format_visibility(item.vis),
+                             format_unsafety(unsafety),
+                             item.ident);
+
+        result.push_str(&header);
+
+        // TODO: Add max_width checking
+        // let budget = try_opt!(context.config.max_width.checked_sub(result.len()));
+        // let indent = offset + result.len();
+
+        let body_lo = context.codemap.span_after(item.span, "{");
+
+        let generics_str = try_opt!(rewrite_generics(context,
+                                                     generics,
+                                                     offset,
+                                                     context.config.max_width,
+                                                     offset + result.len(),
+                                                     mk_sp(item.span.lo, body_lo)));
+        result.push_str(&generics_str);
+
+        let trait_bound_str = try_opt!(rewrite_trait_bounds(context,
+                                                            type_param_bounds,
+                                                            offset,
+                                                            context.config.max_width));
+
+        if offset.width() + result.len() + trait_bound_str.len() > context.config.max_width {
```

Comment:
Could you add a comment here and elsewhere explaining where we are in the output text and why you might break here. (Also elsewhere in this function).